theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -781,3 +783,77 @@ def test_compare_1D_and_2D_upsampling_values(self):
         f_1D = theano.function([], mat_1D, mode=self.compile_mode)
         f_2D = theano.function([], mat_2D, mode=self.compile_mode)
         utt.assert_allclose(f_1D(), f_2D(), rtol=1e-06)
+
+
+def test_conv2d_grad_wrt_inputs():
+
+    inputs_shapes = [(8, 1, 12, 12), (8, 1, 18, 18), (2, 1, 4, 4),
+                     (6, 1, 10, 11), (2, 1, 6, 5), (1, 5, 9, 9)]
+    filters_shapes = [(5, 1, 2, 2), (4, 1, 3, 3), (2, 1, 3, 3),
+                      (1, 1, 2, 5), (4, 1, 2, 2), (4, 5, 2, 2)]
```

Comment:
You do not have to test all of these shapes, but it would be good to have different (in_shape, fltr_shape) combinations that give the same output shape when ss=(2, 2).
For instance, if `fltr_shape = (1, 1, 3, 3)`, then `in_shape = (1, 1, 5, 5)`, `(1, 1, 5, 6)` and `(1, 1, 6, 6)` will all give an output shape of `(1, 1, 2, 2)` in valid mode when ss = (2, 2).